src/Clients/GenericTorrentClient.php

```diff
@@ -1,140 +1,117 @@
 <?php
 
+namespace KeepersTeam\Webtlo\Clients;
+
+use CurlHandle;
+use KeepersTeam\Webtlo\Config\Timeout;
+use KeepersTeam\Webtlo\Config\V0\TorrentClientConfig;
+use Psr\Log\LoggerInterface;
+
 /**
  * Class TorrentClient
  * Базовый класс для всех торрент-клиентов
  */
-abstract class TorrentClient
+abstract class GenericTorrentClient
 {
-    /**
-     * @var string
-     */
-    protected static $base;
-
-    /**
-     * @var string
-     */
-    protected $scheme;
-
-    /**
-     * @var string
-     */
-    protected $host;
-
-    /**
-     * @var string
-     */
-    protected $port;
-
-    /**
-     * @var string
-     */
-    protected $login;
-
-    /**
-     * @var string
-     */
-    protected $password;
-
-    /**
-     * @var string Session ID, полученный от торрент-клиента
-     */
-    protected $sid;
-
-    /**
-     * @var CurlHandle
-     */
-    protected $ch;
+    protected static string $base;
+    protected readonly string $scheme;
+    protected readonly string $host;
+    protected readonly int $port;
+    protected readonly string $login;
+    protected readonly string $password;
+    protected string $sid;
+    protected CurlHandle $ch;
+    protected readonly Timeout $timeout;
 
     /**
      * default constructor
-     * @param bool|int $ssl
-     * @param string $host
-     * @param string $port
-     * @param string $login
-     * @param string $password
      */
-    public function __construct($ssl, $host, $port, $login = '', $password = '')
+    public function __construct(protected readonly LoggerInterface $logger, TorrentClientConfig $config)
```

Comment:
Cant find any `$logger` usage here. I miss something?

Reply:
Used in a inherited clients.

Reply:
But to use it inside client classes, you need to add $logger to it, something like that `$this->logger = $logger;`